frontend/public/components/catalog/catalog-page.tsx

```diff
@@ -411,7 +437,18 @@ export const Catalog = connectToFlags<CatalogProps>(
     coFetch('/api/helm/charts/index.yaml').then(async (res) => {
       const yaml = await res.text();
       const json = safeLoad(yaml);
-      setHelmCharts(json.entries);
+      let kubernetesVersion: string = '';
+      try {
+        const response = await k8sVersion();
+        kubernetesVersion = getK8sGitVersion(response) || '-';
+      } catch {
+        kubernetesVersion = 'unknown';
+      }
+      const filteredHelmChartEntries = filterIncompatibleHelmCharts(
+        json.entries,
+        kubernetesVersion,
+      );
```

Comment:
Why do you have the filtering mechanism separately? While normalizing the helm charts we already go through all the helm charts and reduce it into items for catalog. You can re-use that same reduce to do the filtering easily. We also do filtering of chart versions there. Later on we would have so many helm charts that this would come out to be really expensive.

Reply:
I would like to have the filtering based on `kubeVersion` to be along with this logic - https://github.com/openshift/console/pull/5699